### PR TITLE
docs: Update license year to 2024

### PR DESCRIPTION
**Pull request by joshka**
This removes the copyright year in all the LICENSE files.
It also uses the MIT license text with the wrapping from
https://choosealicense.com/licenses/mit/ rather than the more narrow
wrapped text that was previously used, and adds the explicit "MIT
License" header to each file to make this more obvious.

## Motivation

When I opened the license file, I was unsure if this was MIT licensed or another custom license due to the lack of header and non-standard formatting.

## Solution

Use a more standard formatted license, and ensure the header states this is an MIT License.

### CODE REVIEW

File: tokio-macros/LICENSE [hunk 1]
```diff
@@ -1,32 +1,6 @@
...
-PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
-SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
-IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-DEALINGS IN THE SOFTWARE.
-
-The MIT License (MIT)
-
-Copyright (c) 2019 Yoshua Wuyts
```
Comment (taiki-e) [inline tokio-macros/LICENSE:29]: Please do not remove the existing copyright notice.

Comment (joshka) [reply tokio-macros/LICENSE:29]: Apologies - this was inadvertent. I have re-added this and combined the copyright lines

**Comment by Darksonn** [comment]
We may want to just drop the year ...

Thoughts @carllerche ?

**Comment by Darksonn** [comment]
The Rust project itself already did the same.

**Comment by joshka** [comment]
See also https://github.com/tokio-rs/tracing/pull/2922

**Comment by joshka** [comment]
ping @taiki-e

**Review by Darksonn — APPROVED** [review]
Thank you.

**Pull request merged**